### PR TITLE
finish refactor

### PR DESCRIPTION
**Pull request by nsalhuan**
## What?
Refactor node client to be more intuitive and easier to maintain
## Why?
Our SDK was getting confusing and this was the last portion of the FTUE work.

### CODE REVIEW

File: packages/core/src/controllers/user.ts [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+  ValidateTokenAPIResponse,
+  ValidateTokenArguments,
+  ValidateTokenResponse,
+} from '@slashauth/types';
+import { signBody, signQuery } from '../utils/query';
+import { base64Decode, getURLPath } from '../utils/strings';
+import { Controller } from './controller';
+
+// TODO: Need to add:
+// - How do we add a user to an org?
```
Comment (nsalhuan) [inline packages/core/src/controllers/user.ts:28]: added note here

File: packages/core/src/controllers/user.ts [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+export class UserController extends Controller {
+  constructor(
+    client_id: string,
+    client_secret: string,
+    apiClient: rm.RestClient
+  ) {
+    super(client_id, client_secret, apiClient);
+  }
+
+  async validateToken({
```
Comment (nedrocks) [inline packages/core/src/controllers/user.ts:39]: We should consider moving this to a shared package because we use it in at least two places.

Comment (nedrocks) [reply packages/core/src/controllers/user.ts:39]: Not as part of this PR though

File: packages/core/src/controllers/user.ts [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+      let isUserID: boolean = false;
+      if (decodedClaims.sub.startsWith('user.')) {
+        isUserID = true;
+      }
+
+      const getWalletAddress = async (): Promise<string | null> => {
+        if (isUserID) {
+          const user = await this.getUserByID({ userID: decodedClaims.sub });
+          if (user.result?.data.wallet) {
+            return user.result?.data.wallet;
```
Comment (nedrocks) [inline packages/core/src/controllers/user.ts:78]: iirc it should be wallet.default, no?

**Review by nedrocks — APPROVED** [review]

File: packages/core/src/controllers/user.ts (hunk 2 above)
Comment (nsalhuan) [reply packages/core/src/controllers/user.ts:39]: I'm not following. where would you want to place it?

we use it in the two middleware packages and from the direct client. the middleware packages have access to the core package

Comment (nedrocks) [reply packages/core/src/controllers/user.ts:39]: I'm thinking this function could be in some util package that we use across other typescript packages

Comment (nsalhuan) [reply packages/core/src/controllers/user.ts:39]: ah okay. got it

the utils package would need access to the client as well but I see what you mean

**Pull request merged**